grid-client/state/state.go

```diff
@@ -295,6 +296,13 @@ func (st *State) LoadNetworkFromGrid(name string) (znet workloads.ZNet, err erro
 					if err != nil {
 						return workloads.ZNet{}, errors.Wrapf(err, "failed to get network from workload %s", name)
 					}
+
+					if metdata.PublicNodeID == 0 {
+						if err := json.Unmarshal([]byte(wl.Metadata), &metdata); err != nil {
```

Comment:
I think we can implement metadat part in `NewNetworkFromWorkload`

Reply:
it could, but as i understand, the NewNetworkFromWorkload method gathers the relevant information per network workload, and the PublicNodeID is related to the whole network, so it would work yes, but we would be reassigning the same value over and over again for all network workloads.

Reply:
I think it can be parsed from a zos workload so the functions should do it